### PR TITLE
[build-utils] Fix redwood `/api` detection

### PR DESCRIPTION
**Pull request by styfle**
Previously, the zero detection would see js files in the `/api` directory and attempt to use `@vercel/node`. However, this doesn't work with RedwoodJS which uses a different format for [`/api`](https://github.com/vercel/vercel/tree/master/examples/redwoodjs/api) so we must only use `@vercel/redwood` when RedwoodJS is detected.

### CODE REVIEW

File: packages/now-build-utils/src/detect-builders.ts [hunk 1]
```diff
@@ -309,6 +309,13 @@ export async function detectBuilders(
     options
   );
 
+  if (frontendBuilder && framework === 'redwoodjs') {
+    // RedwoodJS uses the /api directory differently so we must
+    // clear any existing builders and only use `@vercel/redwood`.
+    builders.length = 0;
```
Comment (AndyBitz) [inline packages/now-build-utils/src/detect-builders.ts:315]: TIL

File: packages/now-build-utils/src/detect-builders.ts [hunk 2]
```diff
@@ -309,6 +309,13 @@ export async function detectBuilders(
...
+    builders.push(frontendBuilder);
```
Comment (TooTallNate) [inline packages/now-build-utils/src/detect-builders.ts:316]: Do we need to print a warning here?

**Review by AndyBitz — APPROVED** [review]

**Review by TooTallNate — APPROVED** [review]
Code looks good.

File: packages/now-build-utils/src/detect-builders.ts (hunk 2 above)
Comment (styfle) [reply packages/now-build-utils/src/detect-builders.ts:316]: This is not really a warning, this is the expected behavior. I updated the PR description with more info.

File: packages/now-build-utils/src/detect-builders.ts (hunk 1 above)
Comment (styfle) [reply packages/now-build-utils/src/detect-builders.ts:315]: Its a bit of a hack but it avoids changing `const builders` to `let builders` 😅

**Pull request merged**